app/controllers/settings/external_cert_controller.rb

```diff
@@ -355,10 +357,56 @@ def warning_subjectaltname(cert, warning_messages, required_san_array, service_n
   end
   # rubocop:enable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
 
-  # # Placeholder for trust chain validation
-  # def trust_chain_verify(_cert)
-  #   true
-  # end
+  def trust_chain_verify(cert)
+    _trust_chain_verify(cert, _build_system_wide_certs_store)
+  end
+
+  def _build_system_wide_certs_store
+    cert_store = OpenSSL::X509::Store.new
+
+    system_certs = SystemCertificate.all
+    system_certs.each do |system_cert|
+      cert = system_cert.certificate
+      cert_text = cert.certificate
+      cert_obj = read_cert(cert_text)
+      # Certs must pass parsing to get into the SystemCertificate table; so we ignore failing parse
```

Comment:
This is true in the current `update_ca_certificates` on SUSE (and IIRC RedHat), but some other scripts (Ubuntu, as a notable example) just blindly shove in anything with a .crt extension, assuming the files were put there on purpose.    Probably not important given Velum's likely future, but FWIW...

Reply:
It does do the check, so it does ignore bad certs. It just doesn't complain about them. I feel like that is okay so at least this code doesn't continue doing something with an unparseable cert.